src/scripts/main.js

```diff
@@ -0,0 +1,104 @@
+import 'regenerator-runtime/runtime';
+import {
+    getCartCount, setCartCount, renderCartQuantity, setCartItems,
+    addItemCartSession, openCart, closeCart
+} from './cart';
+
+async function getCategories() {
```

Comment:
Please try employ SRP(Single Responsibility Principle) here by keeping data fetching, cache check, and DOM element creation as separate functions